src/main/java/net/fabricmc/loom/configuration/providers/forge/MinecraftPatchedProvider.java

```diff
@@ -409,33 +420,51 @@ public static void accessTransform(Project project, Path input, Path target) thr
 	}
 
 	private void remapPatchedJar(SharedServiceManager serviceManager) throws Exception {
-		logger.lifecycle(":remapping minecraft (TinyRemapper, srg -> official)");
 		Path mcInput = minecraftPatchedIntermediateAtJar;
+		Path mcPatchedAt = minecraftPatchedMojangAtJar;
 		Path mcOutput = minecraftPatchedJar;
 		Path forgeJar = getForgeJar().toPath();
 		Path forgeUserdevJar = getForgeUserdevJar().toPath();
+
+		if (mcPatchedAt != null) {
+			Files.deleteIfExists(mcPatchedAt);
+		}
+
 		Files.deleteIfExists(mcOutput);
 
-		TinyRemapper remapper = buildRemapper(serviceManager, mcInput);
+		if (getExtension().isForge() && getExtension().getForgeProvider().usesMojangAtRuntime()) {
+			logger.lifecycle(":remapping minecraft (TinyRemapper, srg -> mojang)");
+			TinyRemapper remapper = buildRemapper(MappingOption.WITH_SRG, "srg", "mojang", serviceManager, mcInput);
+			remapJar(remapper, serviceManager, mcInput, mcPatchedAt, forgeJar, forgeUserdevJar, false);
+			logger.lifecycle(":remapping minecraft (TinyRemapper, mojang -> official)");
+			TinyRemapper remapper2 = buildRemapper(MappingOption.WITH_MOJANG, "mojang", "official", serviceManager, mcInput);
+			remapJar(remapper2, serviceManager, mcPatchedAt, mcOutput, null, null, true);
```

Comment:
Why is the game remapped twice? Remapping the game is slow, so it should be avoided.